### ISSUE TITLE
Windows flashinfer_jit_cache-0.6.8.post1 appears to ship sampling.dll without Hopper/sm_90a suppor

### ISSUE THREAD

**Issue by rogeroberg** [SystemPanic/flashinfer-windows#5]
Hi!

My goal is to run vllm-windows with DeepSeek on a Hopper system. I have a running version now and will file an improvement/bug on the vllm-windows repo as well. But one of the issues I had was that sampling.dll did not include Hopper / sm_90a, so I had to recompile FlashInfer with Hopper support locally. It would be good if this could be included from the beginning instead.

The Windows flashinfer_jit_cache-0.6.8.post1 wheel appears to ship sampling.dll without Hopper / sm_90a support.

On a Hopper system, vLLM failed at runtime with:

    no kernel image is available for execution on the device

I inspected the AOT sampling.dll from the wheel with:

    cuobjdump -lelf sampling.dll

The DLL only contains cubins for:

    sampling.*.sm_86.cubin
    sampling.*.sm_120.cubin

There is no sm_90a cubin present.

If I disable the packaged AOT cache directory:

    flashinfer_jit_cache/jit_cache/sampling

FlashInfer falls back to local JIT compilation and builds the sampling module with:

    -gencode=arch=compute_90a,code=sm_90a

After that, the direct FlashInfer sampler test works correctly:

    import torch
    import flashinfer.sampling as s

    logits = torch.randn((4, 32000), device="cuda", dtype=torch.float16)
    top_k = torch.full((4,), 10, device="cuda", dtype=torch.int32)

    out = s.top_k_mask_logits(logits, top_k)
    torch.cuda.synchronize()

    print("OK", out.shape, out.dtype, out.device)

Output:

    OK torch.Size([4, 32000]) torch.float16 cuda:0

So the issue seems to be that the prebuilt Windows AOT sampling.dll in flashinfer-jit-cache was built without Hopper / sm_90a, while runtime JIT works when it targets sm_90a.

Could you include Hopper / sm_90a in the Windows flashinfer-jit-cache build, for example by including 9.0a in FLASHINFER_CUDA_ARCH_LIST?

For example:

    set FLASHINFER_CUDA_ARCH_LIST=8.6 9.0a 12.0f

That should allow the packaged AOT sampling.dll to work on Hopper systems without requiring runtime JIT or manually disabling the packaged sampling cache.

**Comment by rogeroberg** [comment]
Realized now that my issue is that when building vllm-windows for a different arch it still fetches the prebuilt flashinfer-windows wheel. So if you need another arch like me you have to build flashinfer and in install it manually as well. So if you don't want this wheel to include all arch's you can at least update the build instruction for vllm-windows with this.

**Comment by SystemPanic** [comment]
Yes, when you need distinct archs than the prebuilt ones in the release (currently 86 and 120), you need to build from sources both, vLLM and FlahsInfer (or uninstall FlashInfer AOT wheel to use JIT).

I thought that was obvious, as prebuilt archs are specified in the release page, but maybe it is not so obvious. I will update vLLM to reflect that in the build documentation.
